### PR TITLE
Add filter payment

### LINKED ISSUE #50: Add Payment filter
Add `filterpaid` to filter the list by payment using a boolean.
eg. `filterpaid true`

### PR DESCRIPTION
**Pull request by BlazeChron**
Closes #50

Adds FilterPaidCommand, parser and associated tests.

Updates the StudentHasPaidPredicate to support Payment class.

Updates PersonBuilder with Payment.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FilterPaidCommand.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ */
+public class FilterPaidCommand extends Command {
+    public static final String COMMAND_WORD = "filterp";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose paid status matches "
+            + "the specified argument and displays them as a list with index numbers.\n"
+            + "Parameters: HASPAID\n"
+            + "Example: " + COMMAND_WORD + " true";
+
+    public static final String MESSAGE_NOT_IMPLEMENTED_YET =
```
Comment (Slipperybara) [inline src/main/java/seedu/address/logic/commands/FilterPaidCommand.java:21]: Is this message still being used?

Comment (BlazeChron) [reply src/main/java/seedu/address/logic/commands/FilterPaidCommand.java:21]: Good catch, it isn't anymore.

**Review by Slipperybara — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/FilterPaidCommand.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.person.StudentHasPaidPredicate;
+
+/**
+ * Finds and lists all persons in address book whose paid status matches the argument.
+ */
+public class FilterPaidCommand extends Command {
+    public static final String COMMAND_WORD = "filterp";
```
Comment (amitha2210) [inline src/main/java/seedu/address/logic/commands/FilterPaidCommand.java:14]: Maybe the command word could be changed to something more intuitive for the user? Maybe something like "filterpaid"

**Pull request merged**